service/controller/catalog/resource/appcatalogentry/create.go

```diff
@@ -347,6 +347,12 @@ func (r *Resource) newAppCatalogEntries(ctx context.Context, cr v1alpha1.Catalog
 			}
 		}
 
+		// If the latest entry isn't compatible with the current provider
+		// we don't create any entries for this app
+		if !isCompatibleProvider(*latestEntryCR, r.provider) {
+			continue
+		}
```

Comment:
I changed the logic here.

It's possible the compatible providers has been added to an app. If so we shouldn't include any entries.

This is needed since we can't easily update the metadata for existing entries of an app.